packages/queued-request-controller/src/QueuedRequestController.ts

```diff
@@ -224,6 +229,9 @@ export class QueuedRequestController extends BaseController<
     );
   }
 
+  // Note: since we're using queueing for multichain requests to start, this flush could incorrectly flush
+  // multichain requests if the user switches networks on a dapp while multichain request is in the queue.
+  // we intend to remove queueing for multichain requests in the future, so for now we have to live with this.
```

Comment:
I believe this would only be possible via a `wallet_switchEthereumChain` call from the EIP-1193 API since the network selector will not be accessible while there are pending confirmations? Just confirming

Reply:
Yep that sounds right. Still applies. But as I try to make clear this whole approach is a stepping stone and flushings queued requests is an edge case anyways so not going to worry about it too much right now?